modules/runner/src/integrationTest/java/org/apache/ignite/internal/runner/app/jdbc/ItJdbcErrorsAbstractSelfTest.java

```diff
@@ -0,0 +1,626 @@
+                ResultSet rs = stmt.executeQuery();
+
+                rs.next();
+
+                rs.getLong(1);
+            }
+        }, CONVERSION_FAILED, "Cannot convert to long");
+    }
+
+    /**
+     * Test error code for the case when user attempts to get {@code float} value from column whose
+     * value can't be converted to an {@code float}.
+     */
+    @Test
+    public void testInvalidFloatFormat() {
+        checkErrorState(() -> {
+            try (PreparedStatement stmt = conn.prepareStatement("SELECT 'zzz'")) {
+                ResultSet rs = stmt.executeQuery();
+
+                rs.next();
+
+                rs.getFloat(1);
+            }
+        }, CONVERSION_FAILED, "Cannot convert to float");
+    }
+
+    /**
+     * Test error code for the case when user attempts to get {@code double} value from column whose
+     * value can't be converted to an {@code double}.
+     */
+    @Test
+    public void testInvalidDoubleFormat() {
+        checkErrorState(() -> {
+            try (PreparedStatement stmt = conn.prepareStatement("SELECT 'zzz'")) {
+                ResultSet rs = stmt.executeQuery();
+
+                rs.next();
+
+                rs.getDouble(1);
+            }
+        }, CONVERSION_FAILED, "Cannot convert to double");
+    }
+
+    /**
+     * Test error code for the case when user attempts to get {@code byte} value from column whose
+     * value can't be converted to an {@code byte}.
+     */
+    @Test
+    public void testInvalidByteFormat() {
+        checkErrorState(() -> {
+            try (PreparedStatement stmt = conn.prepareStatement("SELECT 'zzz'")) {
+                ResultSet rs = stmt.executeQuery();
+
+                rs.next();
+
+                rs.getByte(1);
+            }
+        }, CONVERSION_FAILED, "Cannot convert to byte");
+    }
+
+    /**
+     * Test error code for the case when user attempts to get {@code short} value from column whose
+     * value can't be converted to an {@code short}.
+     */
+    @Test
+    public void testInvalidShortFormat() {
+        checkErrorState(() -> {
+            try (PreparedStatement stmt = conn.prepareStatement("SELECT 'zzz'")) {
+                ResultSet rs = stmt.executeQuery();
+
+                rs.next();
+
+                rs.getShort(1);
+            }
+        }, CONVERSION_FAILED, "Cannot convert to short");
+    }
+
+    /**
+     * Test error code for the case when user attempts to get {@code BigDecimal} value from column
+     * whose value can't be converted to an {@code BigDecimal}.
+     */
+    @Test
+    public void testInvalidBigDecimalFormat() {
+        checkErrorState(() -> {
+            try (PreparedStatement stmt = conn.prepareStatement("SELECT 'zzz'")) {
+                ResultSet rs = stmt.executeQuery();
+
+                rs.next();
+
+                rs.getBigDecimal(1);
+            }
+        }, CONVERSION_FAILED, "Cannot convert to");
```

Comment:
it looks as we check both code and message for thrown exception, but it not so. Perhaps need to check message too

Reply:
Good catch! fixed